### PR TITLE
[MSHARED-988] Sanitize dependencies

### PR DESCRIPTION
**Pull request by cstamas**
https://issues.apache.org/jira/browse/MSHARED-988

Double artifacts (org.eclipse.aether vs maven-resolver) and other.
Removed them and it just works.

Reordered properties, removed superfluous dependencies.
Used snapshot version for maven-shared-utils:3.3.5-SNAPSHOT

### CODE REVIEW

**Comment by cstamas** [comment]
Setting maven-* artifacts to provided may affect consumers, so I may revert. Still, the presence of maven-plugin-api _assumes_ this artifact is consumed *only by plugins*?

File: pom.xml [hunk 1]
```diff
@@ -55,9 +55,9 @@
   </distributionManagement>
 
   <properties>
-    <maven.version>3.1.1</maven.version>
     <javaVersion>7</javaVersion>
-    <aether.version>0.9.0.M2</aether.version>
+    <maven.version>3.1.1</maven.version>
+    <resolver.version>1.6.2</resolver.version>
```
Comment (slachiewicz) [inline pom.xml:60]: I'm not sure about minimal resolver version here

**Review by slachiewicz — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (cstamas) [reply pom.xml:60]: resolver should be binary compatible, at least the used bits, but we can lower if really needed (hopefully not)

File: pom.xml [hunk 2]
```diff
@@ -98,59 +98,45 @@
...
-      <version>1.4.2</version>
+      <version>${resolver.version}</version>
       <scope>provided</scope>
     </dependency>
 
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-shared-utils</artifactId>
-      <version>3.3.3</version>
+      <version>3.3.5-SNAPSHOT</version>
```
Comment (elharo) [inline pom.xml:139]: no dependencies on snapshot versions please

File: src/main/java/org/apache/maven/shared/artifact/filter/ScopeArtifactFilter.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.artifact.DefaultArtifact;
 import org.apache.maven.artifact.resolver.filter.ArtifactFilter;
-import org.codehaus.plexus.logging.Logger;
+import org.slf4j.Logger;
```
Comment (elharo) [inline src/main/java/org/apache/maven/shared/artifact/filter/ScopeArtifactFilter.java:28]: can this change be separated into its own PR?

**Review by elharo — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (cstamas) [reply pom.xml:139]: Sure, just FTR this PR depends on https://github.com/apache/maven-shared-utils/pull/87

File: src/main/java/org/apache/maven/shared/artifact/filter/ScopeArtifactFilter.java (hunk 3 above)
Comment (cstamas) [reply src/main/java/org/apache/maven/shared/artifact/filter/ScopeArtifactFilter.java:28]: https://github.com/apache/maven-common-artifact-filters/pull/18

File: pom.xml [hunk 4]
```diff
@@ -98,59 +98,51 @@
...
-      <version>1.4.2</version>
+      <version>${resolver.version}</version>
       <scope>provided</scope>
     </dependency>
 
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-shared-utils</artifactId>
-      <version>3.3.3</version>
+      <version>4.0.0-SNAPSHOT</version>
```
Comment (cstamas) [inline pom.xml:145]: @elharo I still used snapshot, as IMO this PR should wait for https://github.com/apache/maven-shared-utils/pull/87

**Comment by cstamas** [comment]
superseded by these
* https://github.com/apache/maven-common-artifact-filters/pull/18
* https://github.com/apache/maven-common-artifact-filters/pull/20

**Comment by jira-importer** [comment]
Resolve #73
